Bash test automation

aahajj opened the pull request:
I've added a bash script to the project that should make testing easier for users on Linux and Mac systems. This script can help automate the testing process and make it more efficient for users.

I didn't update the instructions in the README file. Therefore, maintainers will need to update the instructions if they want to make sure users are aware of this new testing option.

aahajj committed:
add bash script for testing

aahajj committed:
Use python3 for virtual environment

rugo commented:
Kek, nice. I think this script can definitely be useful for students 👍, but I'd like to see two changes:

* Please move the script into a "scripts" subfolder. I think in the main folder it looks a bit "lost". Also the name `run_tests.sh` seems more idiomatic. So Please move it to `scripts/run_tests.sh`.
* Add a brief description of the script to the README

Optionally I'd suggest to add command line arguments:

* It would be great if it was possible to call the script like this to run **all** tests of week1: `./run_tests.sh 1`. That way it would be possible to  include the script in CI pipelines 🕶️.

aahajj committed:
move renamed testing script to scripts directory

aahajj committed:
modify README.md

aahajj committed:
modify README.md

aahajj commented:
Thank you for the suggestions! I have already moved the script to the "scripts" subfolder and renamed it to "run_tests.sh". I have also added a brief description of the script to the README. Not sure if you will like the new README and that it why i did not want to modify it at first lol.

Regarding the command line arguments, I have decided not to implement them since users may not always know which class number to run in advance. The script already provides the option to choose the class number during runtime, which I believe is more user-friendly.

aahajj committed:
remove redundant line instaling pytest

aahajj committed:
add cmd script for windows

aahajj committed:
update README.me

aahajj commented:
There is a a script for windows users too! :))

rugo commented on README.md:
```diff
@@ -4,42 +4,16 @@ As part of the lecture, students have to implement the missing chunks.
+Only one package is required as an external dependency. That is `pytest` which is required to run the (you guessed it) tests. To install pytest just run ``pip install pytest`` on your terminal.
 
-Please install pytest only via the Python package manager `pip`. In an own virtual environment this can be achieved with:
+## testing 
```
Thanks. But the manual methods should still be described.

Please add your `## testing` section as last section to the README, aka:

## Setup
[...]
## Code and Tasks
[...]

## Test Code
[...]

## Build and Test Scripts
$YOUR_SECTION

rugo commented:
Hi again. Thanks for the additions!

I have one last wish and then this PR is ready for merging: https://github.com/cry-college/Cry.College/pull/5/files#r1182289105

Greets
RuGo
![dwight](https://user-images.githubusercontent.com/9132874/235626547-22c21658-5569-4fb3-a7e5-6bf2153c9016.gif)

aahajj committed:
Merge remote-tracking branch 'upstream/main' into bash-test-automation

aahajj committed:
update readme

aahajj commented:
Done :))

rugo commented on README.md:
````diff
@@ -36,16 +36,22 @@ In case you don't want to set an environment variable, you can configure [vscode
-```bash
-pytest CryCollege/week3/weierstrass_curve.py 
-```
+```bashpytest CryCollege/week3/weierstrass_curve.py``` 
````
There is a missing newline. This won't render properly. Add a newline after `bash`.

rugo commented:
Thanks, but there is one last flaw in the README. The markdown is broken due to a missing newline (see https://github.com/cry-college/Cry.College/pull/5/files#r1185907263). Everything else looks great.

aahajj committed:
Add missing new line in README.md

aahajj commented:
oops! now it is done fr 😅

rugo commented:
Great PR, thanks.

![thumbs-thumbs-up-kid](https://user-images.githubusercontent.com/9132874/236485824-ad864d9d-cefe-4a80-9925-88668e2aa302.gif)
